xbmc/platform/darwin/ios-common/network/NetworkIOS.mm

```diff
@@ -0,0 +1,403 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#import "NetworkIOS.h"
+
+#import "utils/StringUtils.h"
+#import "utils/log.h"
+
+#import "platform/darwin/ios-common/network/route.h"
+
+#import <array>
+
+#import <arpa/inet.h>
+#import <ifaddrs.h>
+#import <net/if.h>
+#import <net/if_dl.h>
+#import <netdb.h>
+#import <netinet/in.h>
+#import <netinet6/in6.h>
+#import <resolv.h>
+#import <sys/ioctl.h>
+#import <sys/socket.h>
+#import <sys/sockio.h>
+#import <sys/sysctl.h>
+
+CNetworkInterfaceIOS::CNetworkInterfaceIOS(CNetworkIOS* network, std::string interfaceName)
+  : m_interfaceName(interfaceName)
+  , m_network(network)
+{
+}
+
+CNetworkInterfaceIOS::~CNetworkInterfaceIOS() = default;
+
+std::string CNetworkInterfaceIOS::GetInterfaceName() const
+{
+  return m_interfaceName;
+}
+
+bool CNetworkInterfaceIOS::IsEnabled() const
+{
+  struct ifreq ifr;
+  strcpy(ifr.ifr_name, m_interfaceName.c_str());
+  if (ioctl(m_network->GetSocket(), SIOCGIFFLAGS, &ifr) < 0)
+    return false;
+
+  return ((ifr.ifr_flags & IFF_UP) == IFF_UP);
+}
+
+bool CNetworkInterfaceIOS::IsConnected() const
+{
+  struct ifaddrs* interfaces = nullptr;
+
+  if (getifaddrs(&interfaces) != 0)
+    return false;
+
+  for (struct ifaddrs* iface = interfaces; iface != nullptr; iface = iface->ifa_next)
+  {
+    if (StringUtils::StartsWith(iface->ifa_name, m_interfaceName))
+    {
+      if ((iface->ifa_flags & (IFF_UP | IFF_RUNNING)) == (IFF_UP | IFF_RUNNING) &&
+          iface->ifa_dstaddr != nullptr)
+      {
+        if (interfaces != nullptr)
+          freeifaddrs(interfaces);
+        return true;
+      }
+    }
+  }
+
+  if (interfaces != nullptr)
+    freeifaddrs(interfaces);
+
+  return false;
+}
+
+std::string CNetworkInterfaceIOS::GetMacAddress() const
+{
+  return "";
```

Comment:
maybe add a `//! @todo`

Reply:
It is not possible to retrieve a mac address on an ios device. There were workarounds prior to ios11, but these are no longer valid, and was actually deprecated in ios 7.

Reply:
This function isn't correctly implemented and that should be documented somehow.
A todo comment doesn't care if it is actually possible or not, only that it should be done.
Maybe it can be done in future versions again.

Reply:
happy to comment it, but the likelihood of it coming back after tried to stamp it out for the better part of 6 years by apple so far is probably pretty unlikely.

Reply:
Are the comments ok? Im not always the best at clear concise commenting